### ISSUE TITLE
typecheck-number to general

### ISSUE THREAD

**Issue by marcus13371337** [Gillespie59/eslint-plugin-angular#562]
The typecheck-number rule is reacting to my code, when I'm checking if a input tag is of type number.
`type === 'number'`

Should only react on:
`typeof type === 'number'`

**Comment by Gillespie59** [comment]
That's normal. 

If the developer use the next snippet, ESLint should throw an error

```javascript
const type = typeof variable
type === 'number'
```

https://github.com/Gillespie59/eslint-plugin-angular/blob/master/test/typecheck-number.js#L37

**Comment by marcus13371337** [comment]
No because now I use it like this:
`let input = document.findByClassName('my-input')
  let type = input.attrs.type
  if(type === 'number') return true
`

I don't want to use angular.isNumber() here :)

**Comment by Gillespie59** [comment]
Could you send a PR with a failing test ?
